tango_ros_common/tango_ros_native/src/occupancy_grid_file_io.cpp

```diff
@@ -45,6 +85,13 @@ bool SaveOccupancyGridDataToPgmFile(
     const nav_msgs::OccupancyGrid& occupancy_grid) {
   std::string map_directory_with_trailing_slash = map_directory;
   AddTrailingSlashToDirectoryPathIfNeeded(map_directory_with_trailing_slash);
+  if (!DirectoryPathExists(map_directory_with_trailing_slash)) {
+    LOG(INFO) << "Directory " << map_directory_with_trailing_slash << " does not exist, creating it now.";
+    if (MakeDirectoryPath(map_directory_with_trailing_slash, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0) {
```

Comment:
Easier if MakeDirectoryPath returns a Boolean. It is not directly clear from outside the function what the integers mean in the specific implementation if this function.